### PR TITLE
Retain whitespace before color().  Fix for #26.

### PR DESCRIPTION
**Pull request by studiosciences**
Replace balance-matched with postcss-value-parser to fix issue whitespace removal issue #26.

### CODE REVIEW

**Comment by studiosciences** [comment]
I also have this working with postcss-value-parser in a separate branch:
https://github.com/studiosciences/postcss-color-function/blob/e91a9d7f72e660699d46e1b968918575cc29e620/index.js

However, I wasn't 100% sure that it was ok to replace the node in the way I am...
1. Change the type to word.
2. Insert the rgba() as value.
3. Delete the child nodes array.

If you want me to update the Pull Request to that instead, let me know.

**Comment by studiosciences** [comment]
Merged in the branch that used postcss-value-parser.

Minor Issue: We call .toString() even if no actual color functions were found. This would only occur if there was a "color(" in the value that was not actually a color function, such as "hover-color(".

File: index.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ module.exports = postcss.plugin("postcss-color-function", function() {
       }
 
       decl.value = helpers.try(function transformColorValue() {
```
Comment (studiosciences) [inline index.js:19]: This try block and helper seems irrelevant now.

Comment (TrySound) [inline index.js:19]: color may throw an error.

Comment (TrySound) [inline index.js:19]: It would be cool to catch and warn separately.

Comment (TrySound) [inline index.js:19]: But in the next pr I think.

Comment (studiosciences) [inline index.js:19]: There is a test for throwing a readable error that passes. I assume it passes because colorFn.convert() is the one that throws the error.

There had been a throw error on a mismatched bracket, but that seems irrelevant with postcss-value-parser.

**Comment by studiosciences** [comment]
Are you good to merge then? Version number needs a bump.

**Comment by MoOx** [comment]
@studiosciences can you squash your commit (especially the lasts)?
Also why version number needs a bump? According to tests, it's a bugfix.

**Comment by studiosciences** [comment]
@MoOx I haven't squashed before, but I'll see what I can do.

I'll leave any version changing to you.

**Comment by studiosciences** [comment]
Squashing complete. It appears as one commit in my branch, but the Pull Request lists six.

Not sure how to fix that or if anything needs to be fixed.

**Comment by MoOx** [comment]
There is one commit, just refresh the page :D

**Comment by studiosciences** [comment]
Weird. I saw the squashed commit as an additional commit. Looks right now.

**Comment by MoOx** [comment]
When you push a commit, github update your open PR, but if you force push, do not remove previous commit until your reload. Pretty buggy UX.

**Comment by MoOx** [comment]
Will merge and release tomorrow, I have to go :)

**Comment by studiosciences** [comment]
Anything more I could do to wrap this up?

**Comment by MoOx** [comment]
Oh sorry, I forgot. Let me handle this this evening.

**Comment by MoOx** [comment]
Released as 2.0.1, thanks!
